### ISSUE TITLE
Hänger sig?

### ISSUE THREAD

**Issue by magnusvmt** [freedrikp/EDA095-Project#3]
Hänger den sig en liten bit in i videon för er också? Jag la nämligen till ljud och trodde att det var jag som gjort något fel med alla buffers och skit. Men det verkar vara samma med orginalet för mig.

**Comment by Nisseman3000** [comment]
Om du menar att servern dör så är det nog "monitor.closeIt()" som anropas fel så de e bara att kommentera ut den raden så ska filmen nog rulla på. Jag gör en commit med fixet

**Comment by magnusvmt** [comment]
Fungerar fortfarande inte för mig. Den hänger sig när ökenscenen börjar ungefär. Verkar som att Servern kastar: java.lang.OutOfMemoryError: Java heap space.

**Comment by freedrikp** [comment]
Jag har fixat buggen med att socketen stänger sig. Japp java får slut på sin heap men det kör på ändå det ända som händer är att datorn flyttar fokus från vårt gui till eclipse, bara att ta fram vårt gui så ser du att den fortfarande kör.

**Comment by magnusvmt** [comment]
Nej, det är inte det. Det måste vara något fel här hos mig helt enkelt. Den stannar alltid uppspelningen, sen spelar det ingen roll vad jag gör, den vägrar fortsätta. Versionen som bara spelar upp lokalt fungerar fint. Väldigt konstigt detta.

**Comment by freedrikp** [comment]
Hmm, jag vet vad du menar har hänt mig också,la till så att efter sista framen så skickas -1 så alla frames skickas faktiskt, för den slutar som den ska. Progressbaren slutar att updateras vid 101 % varför det är så vet jag inte men det spelas fortfarande upp efter det.

**Comment by magnusvmt** [comment]
Har inte fungerat en enda gång hemma, jag försöker fixa ljudet i skolan på måndag istället helt enkelt, förhoppningsvis fungerar det där. Jag lyckades egentligen i onsdags, men det fungerade bara i ett par sekunder eftersom allt stannar.

**Comment by arturmatulaniec** [comment]
Ehm, liten update tack... Jag startade möget hemma idag, det verkar ju hålla ett bra tempo hela tiden? Paus och play funkar, dock verkar ju den progressbaren fungera något galet (det räknades upp till typ 2000% och sen ned till 0 igen) så tänkte kika på det nu. Lite nya klasser har kommit till så tar mig igenom dem och läser vad som händer. 

Annars börjar det se riktigt grymt ut! :) 

EDIT: 
Såg att progressbaren går efter antalet frames, nevermind. Försöker komma på nått bra att göra för att göra att det går från 0->100.

**Comment by freedrikp** [comment]
hmm, ja det verkar hänga sig med större filer :(

**Comment by freedrikp** [comment]
Asså progressbaren funkar så du behöver inte pilla på den, anledningen är att vi har en buffert i klienten som har 100 frames

**Comment by freedrikp** [comment]
Jag har fixat så att frames spelas upp i rätt tempo, det gjorde jag i torsdags tror jag.

**Comment by arturmatulaniec** [comment]
Ok ok, jag kikar lite och så tittar på det med paus över nätet, för nu e det bara lokalt som du säger.

**Comment by freedrikp** [comment]
Okej fast nu när vi har en buffert i klienten tycker jag nog att play/pause ska vara lokalt. Har inte pratat med nån annan men jag känner att det bör fungera så och att vi lägger till ytterliggare en knapp som stoppar streamen. Tänkte vi kunde snacka om detta när vi träffas nästa gång.

**Comment by arturmatulaniec** [comment]
Yesbox, jag måste säga att det blir nog bättre, rent intuitivt så blir det ju så att servern skickar och klienten spelar upp det den får. Ändrar utskriften så man ser vad som menas istället, för tycker in % är bra i detta fallet om vi kör lokalt. 

men vad ska annars göras? Jag sitter för tillfället och försöker få mig en överblick över alla nya klasser men ser inget som står på tur härnäst att göras? Det vi sa vi skulle fixa har vi ju redan gjort...

**Comment by freedrikp** [comment]
Ja det kan du göra. Om du har lust kan du göra så att 100 frames inte är hårdkodat varken i gui:t eller i buffern. Nä annars finns det inte mer. Jag har tjuvstartat lite med protokollet och förberett för det men vågar inte göra mer nu. Vi får se hur det går för magnus också med ljudet.

**Comment by arturmatulaniec** [comment]
Yes, har sett och pillat lite i GUIt. Dubbelklick nu ger fullscreen (dock inte på bilden, det med JLabel imageIcon e sjukt komplicerat att få att bli fullscreen). Samt att enkelklick på filmen pausar och startar filmen igen. 

Annars har jag bara fått koll på allt nytt som kommit in! Testade en stor fil, hahaha, gud vad långsamt det går... Däremot så spelar den ju men ja, det känns som om man sitter på ett 56kb modem :d

**Comment by freedrikp** [comment]
haha nice, ja servern blir seg på stora filer. Har du committat?

**Comment by arturmatulaniec** [comment]
yes, gjorde visst bara en commit och inte commit&push... men fixat och nu ska det vara uppe!

**Comment by freedrikp** [comment]
hehe lätt hänt

**Comment by freedrikp** [comment]
Jag har gjort en klass configuration där konstanter kan definieras, denna klassen skulle kunna parsa en config fil i framtiden. Plus att jag la till så att bilderna komprimeras med en valfri faktor innan dom skickas över till klienten, blev typ 90% mindre bytes nu än innan haha.
